### PR TITLE
rapidyaml: add version 0.7.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **rapidyaml/0.7.0**

https://github.com/biojppm/rapidyaml/compare/v0.5.0...v0.7.0

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rapidyaml/all/conanfile.py [hunk 1]
```diff
@@ -21,12 +21,16 @@ class RapidYAMLConan(ConanFile):
...
         "with_tab_tokens": [True, False],
+        "with_default_callback_uses_exceptions": [True, False],
+        "with_assert": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "with_default_callbacks": True,
         "with_tab_tokens": False,
+        "with_default_callback_uses_exceptions": False,
```
Comment (RubenRBS) [inline recipes/rapidyaml/all/conanfile.py:32]: Afaik this one should only be valid if `with_default_callbacks` is `true` as per https://github.com/biojppm/rapidyaml/compare/v0.5.0...v0.6.0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR17-R19 - we might need to delete the option in `configure` if that is true

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot :)

Minor comment about the handling of the new option, otherwise looks good

**Review by RubenRBS — APPROVED** [review]
Thanks! No worries about the late response at all :)

**Review by uilianries — APPROVED** [review]
LGTM https://github.com/biojppm/rapidyaml/compare/v0.5.0...v0.7.0

**Pull request merged**